esp-wifi/src/wifi/mod.rs

```diff
@@ -2610,8 +2712,8 @@ impl WifiController<'_> {
     /// A blocking wifi network scan with default scanning options.
     pub fn scan_n<const N: usize>(
         &mut self,
-    ) -> Result<(heapless::Vec<AccessPointInfo, N>, usize), WifiError> {
-        self.scan_with_config_sync(Default::default())
+    ) -> Result<(alloc::vec::Vec<AccessPointInfo>, usize), WifiError> {
```

Comment:
- We no longer need these methods to be const-generic
- We should return an ApList or similar custom struct, and name the fields, I had to look up (from the code, thanks past me...) what the `usize` was meant to be.
- Do we have a "just give me all" API, besides `scan_results`?

Reply:
Yes we don't strictly need const-generics anymore - I tried to avoid reworking that all at once but since you called it out let's look into it now